pkg/network/kubemacpool.go

```diff
@@ -87,9 +87,10 @@ func fillDefaultsKubeMacPool(conf, previous *opv1alpha1.NetworkAddonsConfigSpec)
 }
 
 func changeSafeKubeMacPool(prev, next *opv1alpha1.NetworkAddonsConfigSpec) []error {
-	if prev.KubeMacPool != nil && !reflect.DeepEqual(prev.KubeMacPool, next.KubeMacPool) {
+	if prev.KubeMacPool != nil && next.KubeMacPool != nil && !reflect.DeepEqual(prev.KubeMacPool, next.KubeMacPool) {
```

Comment:
is it fine to just add kmp to the test https://github.com/kubevirt/cluster-network-addons-operator/blob/master/test/e2e/workflow/validation_test.go

configSpec add kmp, and then it will remove it ?
(did same there for nmstate)